lib/middleman.js

```diff
@@ -0,0 +1,58 @@
+'use strict'
+
+const reusify = require('reusify')
+
+function middleman (complete) {
+  var functions = []
+  var hasMiddlewares = false
+  var pool = reusify(Holder)
+
+  return {
+    use,
+    run
+  }
+
+  function use (f) {
+    hasMiddlewares = true
+    functions.push(f)
+    return this
+  }
+
+  function run (req, res) {
+    if (!hasMiddlewares) {
+      complete(null, req, res)
+      return
+    }
+
+    const holder = pool.get()
+    holder.req = req
+    holder.res = res
+    holder.done()
+  }
+
+  function Holder () {
+    this.next = null
+    this.req = null
+    this.res = null
+    this.i = 0
+
+    var that = this
+    this.done = function (err) {
```

Comment:
Instead of calling this `done`, could make sense call it `next`?
Correct me if I am wrong, but I'm asking because at the end of the day when a user calls the `next()` function, he is calling this one.